go/private/go_toolchain.bzl

```diff
@@ -52,34 +51,62 @@ def _get_linker():
-    "tools": attr.label(mandatory = True, allow_files = True),
-    "stdlib": attr.label(mandatory = True, allow_files = True),
-    "headers": attr.label(mandatory = True),
-    "link_flags": attr.string_list(default=[]),
-    "cgo_link_flags": attr.string_list(default=[]),
-    "goos": attr.string(mandatory = True),
-    "goarch": attr.string(mandatory = True),
-    "_external_linker": attr.label(default=_get_linker),
-}
+def _asm(bootstrap):
+  if bootstrap:
+    return None
+  return Label("//go/tools/builders:asm")
 
-go_toolchain_attrs = go_toolchain_core_attrs + {
-    "asm": attr.label(allow_files = True, single_file = True, executable = True, cfg = "host", default=Label("//go/tools/builders:asm")),
-    "compile": attr.label(allow_files = True, single_file = True, executable = True, cfg = "host", default=Label("//go/tools/builders:compile")),
-    "link": attr.label(allow_files = True, single_file = True, executable = True, cfg = "host", default=Label("//go/tools/builders:link")),
-    "cgo": attr.label(allow_files = True, single_file = True, executable = True, cfg = "host", default=Label("//go/tools/builders:cgo")),
-    "test_generator": attr.label(allow_files = True, single_file = True, executable = True, cfg = "host", default=Label("//go/tools/builders:generate_test_main")),
-    "extract_package": attr.label(allow_files = True, single_file = True, executable = True, cfg = "host", default=Label("//go/tools/extract_package")),
-    "crosstool": attr.label(default=Label("//tools/defaults:crosstool")),
-    "_go_toolchain_flags": attr.label(default=Label("@io_bazel_rules_go//go/private:go_toolchain_flags")),
-}
+def _compile(bootstrap):
+  if bootstrap:
+    return None
+  return Label("//go/tools/builders:compile")
+
+def _link(bootstrap):
+  if bootstrap:
+    return None
+  return Label("//go/tools/builders:link")
+
+def _cgo(bootstrap):
+  if bootstrap:
+    return None
+  return Label("//go/tools/builders:cgo")
+
+def _test_generator(bootstrap):
+  if bootstrap:
+    return None
+  return Label("//go/tools/builders:generate_test_main")
+
+def _extract_package(bootstrap):
+  if bootstrap:
+    return None
+  return Label("//go/tools/extract_package")
 
 go_toolchain = rule(
     _go_toolchain_impl,
-    attrs = go_toolchain_attrs,
+    attrs = {
+        "sdk": attr.string(mandatory = True),
+        "root": attr.label(mandatory = True),
+        "go": attr.label(mandatory = True, allow_files = True, single_file = True, executable = True, cfg = "host"),
+        "tools": attr.label(mandatory = True, allow_files = True),
+        "stdlib": attr.label(mandatory = True, allow_files = True),
+        "headers": attr.label(mandatory = True),
+        "link_flags": attr.string_list(default=[]),
+        "cgo_link_flags": attr.string_list(default=[]),
+        "goos": attr.string(mandatory = True),
+        "goarch": attr.string(mandatory = True),
+        "bootstrap":attr.bool(mandatory = True),
```

Comment:
Nit: space after `:`

Would also be good to add space around the `=` after `default` arguments below.